core/store/src/trie/shard_tries.rs

```diff
@@ -1,104 +1,122 @@
-        TrieUpdate::new(Rc::new(self.get_trie_for_shard(shard_id)), state_root)
+    pub fn new_trie_update(&self, shard_uid: ShardUId, state_root: CryptoHash) -> TrieUpdate {
+        TrieUpdate::new(Rc::new(self.get_trie_for_shard(shard_uid)), state_root)
     }
 
-    pub fn new_trie_update_view(&self, shard_id: ShardId, state_root: CryptoHash) -> TrieUpdate {
-        TrieUpdate::new(Rc::new(self.get_view_trie_for_shard(shard_id)), state_root)
+    pub fn new_trie_update_view(&self, shard_uid: ShardUId, state_root: CryptoHash) -> TrieUpdate {
+        TrieUpdate::new(Rc::new(self.get_view_trie_for_shard(shard_uid)), state_root)
     }
 
-    fn get_trie_for_shard_internal(&self, shard_id: ShardId, is_view: bool) -> Trie {
+    fn get_trie_for_shard_internal(&self, shard_uid: ShardUId, is_view: bool) -> Trie {
         let cache = if is_view {
-            self.view_caches[shard_id as usize].clone()
+            self.view_caches[&shard_uid].clone()
         } else {
-            self.caches[shard_id as usize].clone()
+            self.caches[&shard_uid].clone()
         };
-        let store = Box::new(TrieCachingStorage::new(self.store.clone(), cache, shard_id));
-        Trie::new(store, shard_id)
+        let store = Box::new(TrieCachingStorage::new(self.store.clone(), cache, shard_uid));
+        Trie::new(store, shard_uid)
     }
 
-    pub fn get_trie_for_shard(&self, shard_id: ShardId) -> Trie {
-        self.get_trie_for_shard_internal(shard_id, false)
+    pub fn get_trie_for_shard(&self, shard_uid: ShardUId) -> Trie {
+        self.get_trie_for_shard_internal(shard_uid, false)
     }
 
-    pub fn get_view_trie_for_shard(&self, shard_id: ShardId) -> Trie {
-        self.get_trie_for_shard_internal(shard_id, true)
+    pub fn get_view_trie_for_shard(&self, shard_uid: ShardUId) -> Trie {
+        self.get_trie_for_shard_internal(shard_uid, true)
     }
 
     pub fn get_store(&self) -> Arc<Store> {
         self.store.clone()
     }
 
     pub fn update_cache(&self, transaction: &DBTransaction) -> std::io::Result<()> {
-        let mut shards = vec![Vec::new(); self.caches.len()];
+        let mut shards = self
+            .caches
+            .iter()
+            .map(|(shard_uid, ..)| (*shard_uid, Vec::new()))
+            .collect::<HashMap<_, _>>();
         for op in &transaction.ops {
             match op {
                 DBOp::UpdateRefcount { col, ref key, ref value } if *col == DBCol::ColState => {
-                    let (shard_id, hash) = TrieCachingStorage::get_shard_id_and_hash_from_key(key)?;
-                    shards[shard_id as usize].push((hash, Some(value.clone())));
+                    let (shard_uid, hash) =
+                        TrieCachingStorage::get_shard_uid_and_hash_from_key(key)?;
+                    shards.get_mut(&shard_uid).unwrap().push((hash, Some(value.clone())));
+                    // [shard_uid as usize].push((hash, Some(value.clone())));
```

Comment:
What happened here?

Reply:
This is just some artifacts from me copy-pasting some code. Removed